E2E Suite (#592)
kyma-project/lifecycle-manager: tests/e2e_test/README.md

# E2e Tests

> This Subdirectory contains e2e tests used for E2E Verification.

Reviewer: ```suggestion
This subdirectory contains e2e tests used for the e2e verification.
```
the e2e verification of what? Could we add what is verified?

Reviewer 2: The watcher component is what is being verified. I will add it 👍

> Those tests are written to run inside a pipeline. But they can also be run on a local machine, as described in the next section.

Reviewer: ```suggestion
Those tests are written to run inside a pipeline. But they can also be run on a local machine. See the [Run the e2e tests locally](#run-the-e2e-tests-locally) section for details.
```

> ## Content

Reviewer: ```suggestion
## Contents

The `e2e_test` directory consists of the following elements:
```

> - [Test  Suite](./e2e_suite_test.go): Sets up needed clients and configurations.

Reviewer: ```suggestion
- [Test suite](./e2e_suite_test.go) that sets up required clients and configurations.
```

> - [Watcher end-to-end tests](./e2e_watcher_test.go): Tests the end-to-end workflow of the watcher.

Reviewer: ```suggestion
- [Watcher end-to-end tests](./e2e_watcher_test.go) that includes tests and the end-to-end workflow of the watcher.
```
Should we also list the Makefile here?

Reviewer 2: Yes. Will add it.

> ## Run the Tests locally

Reviewer: ```suggestion
## Run the e2e tests locally
```

> 1. Setup local environment using [this guide](../../docs/developer/local-test-setup.md). Only cluster creation and pre-requisites steps are needed, rest can be skipped. 

Reviewer: ```suggestion
1. Set up your local environment using steps 1 to 3 from [this guide](../../docs/developer-tutorials/local-test-setup.md). Create a cluster, and install Istio CRDs and `cert-manager`. Skip all the remaining steps from the guide.
```

> 2. Export K8s Configurations into environment variables

Reviewer: ```suggestion
2. Export the following  K8s configurations as environment variables:
```

    ```shell
    export KCP_KUBECONFIG=$(k3d kubeconfig write kcp-local)
    export SKR_KUBECONFIG=$(k3d kubeconfig write skr-local)
    ```

> 3. Run `make`

Reviewer: ```suggestion
3. Run `make`.
```
